File: src/lib/types/weight.ts
export interface Weight {
  id: number;
  measuredDate: string;  // ISO date string (YYYY-MM-DD)
  weight: number;
  note: string | null;
  createdAt: string;     // ISO datetime string
}

export interface WeightCreate {
  measuredDate: string;
  weight: number;
  note?: string | null;
}

export interface WeightUpdate {
  measuredDate: string;
  weight: number;
  note?: string | null;
}

File: src-tauri/src/db/models/weight.rs
use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, serde::Serialize, serde::Deserialize)]
#[sea_orm(table_name = "weights")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    #[sea_orm(unique)]
    pub measured_date: Date,
    pub weight: f64,
    pub note: Option<String>,
    pub created_at: DateTime,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
